pkg/cli/cmd/resource/create/create.go

```diff
@@ -0,0 +1,158 @@
+/*
+Copyright 2023 The Radius Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package create
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+	"strings"
+
+	"github.com/radius-project/radius/pkg/cli"
+	"github.com/radius-project/radius/pkg/cli/clients_new/generated"
+	"github.com/radius-project/radius/pkg/cli/clierrors"
+	"github.com/radius-project/radius/pkg/cli/cmd/commonflags"
+	"github.com/radius-project/radius/pkg/cli/cmd/resourceprovider/common"
+	"github.com/radius-project/radius/pkg/cli/connections"
+	"github.com/radius-project/radius/pkg/cli/framework"
+	"github.com/radius-project/radius/pkg/cli/output"
+	"github.com/radius-project/radius/pkg/cli/workspaces"
+	"github.com/spf13/cobra"
+)
+
+// NewCommand creates an instance of the `rad resource create` command and runner.
+func NewCommand(factory framework.Factory) (*cobra.Command, framework.Runner) {
+	runner := NewRunner(factory)
+
+	cmd := &cobra.Command{
+		Use:   "create [resource type] [name] [input]",
+		Short: "Create or update a resource",
+		Long: `Create or update a resource
+		
+Resources are the entities are the primary entities that make up applactions.
+
+Input can be passed in using a file or inline JSON as the second argument. Prefix the input with '@' to indicate a file path.
```

Comment:
We use `@` for bicep parameter files. I think that's the only other place where we accept a filename as an argument/option. 

I'm open to other ideas. We've only got one other point of comparison, so we could make a change pretty easily. 

In case we're considering this because of `kubectl` ..... We should be a little bit careful about adopting another project's conventions unless we're going to follow them faithfully. 

For example:
- `kubectl apply -f ...` does some things that don't really apply here like globbing
- `kubectl apply` has multiple ways to specify the input (see `-k`, `-d`).

Reply:
Yeah I thought of `-f` as it is a shorthand for a file path 

`rad resource create --from-file <filepath>` or `rad resource create -f <filepath>`

Reply:
I'm good with that as long as we're willing to change `rad deploy` to be consistent. I think this could be an improvement to both commands :+1:.

I'll go ahead and do `-f` + `--from-file` on this command. I'm going to keep it really simple while we make all of the detailed decisions that are part of this.

Hint: we're also going to need this for `rad resource-provider create`. 

@Reshrahim can you write up a small spec on the behaviors? 

It should answer questions like this:

- Does it support multiple values?
- Does it accept wildcards?
- Does it support stdin? (what's the syntax?)
- What's the alternative for passing data directly without a file?

Reply:
Discussed offline with @Reshrahim and we're happy with this.

Reply:
Will capture the behaviors of `-f` in the design doc